lib/jsonrpc.ml

```diff
@@ -554,27 +554,41 @@ let get name dict =
     raise (Malformed_method_request str)
   | Some v -> v
 
-let call_of_string str =
+let version_and_call_of_string str =
   match of_string str with
   | Dict d ->
-    let name = match get "method" d with String s -> s | _ -> raise (Malformed_method_request str) in
-    let params =
+    let name =
+      match get "method" d with
+      | String s -> s
+      | _ -> raise (Malformed_method_request "method")
```

Comment:
I was going to say:
```
It allows us to throw a better error - while it's nice to know what went wrong (e.g. "method") it'd also be nice to see the string we were trying to parse.
```
but I now realise we're passing the string in here, so it adds nothing to return it in the exception. In which case, throwing the exception with the thing that went wrong ("method") seems sensible to me.

However, since you're calling `get` rather than `get'`, you're going to get a different type of exception depending on whether `get` fails or returns you something of the wrong type. That should be fixed.